frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/index.tsx

```diff
@@ -227,113 +252,143 @@ export default function SaveExport({
-          >
-            Cancel
-          </DownloadButton>
-          <DownloadButton
-            sdsStyle="square"
-            sdsType="primary"
-            sdsSize="large"
-            onClick={handleDownload}
-            disabled={!selectedFileTypes.length}
-            data-testid="dialog-download-button"
-          >
-            Download
-          </DownloadButton>
-        </StyledButtonContainer>
+
+          <UnderlyingDataChangeBanner width="100%" />
+
+          <StyledButtonContainer>
+            <DownloadButton
+              sdsType="secondary"
+              sdsStyle="minimal"
+              onClick={handleButtonClick}
+            >
+              Cancel
+            </DownloadButton>
+            <DownloadButton
+              sdsStyle="square"
+              sdsType="primary"
+              sdsSize="large"
+              onClick={handleDownload}
+              disabled={!selectedFileTypes.length}
+              data-testid="dialog-download-button"
+            >
+              Download
+            </DownloadButton>
+          </StyledButtonContainer>
+        </StyledModalContent>
       </StyledModal>
     </>
   );
 }
 
-function processSvg({
+function generateSvg({
   svg,
   tissueName,
-  height,
+  heatmapHeight,
+  heatmapWidth,
 }: {
   svg: string;
   tissueName: Tissue;
-  height: number;
+  heatmapHeight: number;
+  heatmapWidth: number;
 }) {
   const heatmapNode = new DOMParser().parseFromString(svg, "image/svg+xml");
   const heatmapContainer = heatmapNode
     .querySelector("foreignObject")
     ?.querySelector("div");
 
+  // This is an svg element created in dom when downloadStatus.isLoading is true
+  const banner = document.getElementById("data-message-banner");
+
+  if (!banner) return "";
+
+  const bannerHeight = banner.clientHeight || 0; // Used to create room for the banner
+  const bannerWidth = banner.clientWidth || 0;
+
   // Render elements to SVG
-  const xAxisSvg = renderXAxis({ heatmapContainer });
-  const yAxisSvg = renderYAxis({ heatmapContainer, height, tissueName });
+  const xAxisSvg = renderXAxis({
+    heatmapContainer,
+    yOffset: bannerHeight,
+  });
+  const yAxisSvg = renderYAxis({
+    heatmapContainer,
+    heatmapHeight,
+    tissueName,
+    yOffset: bannerHeight,
+  });
   const dotsSvg = renderDots({
     heatmapContainer,
     tissueName,
-    xPosition: yAxisSvg!.width.baseVal.value,
+    yOffset: bannerHeight,
+  });
+  const legendSvg = renderLegend({
+    heatmapContainer,
+    yOffset: bannerHeight,
   });
-  const legendSvg = renderLegend({ heatmapContainer });
 
   const svgWidth =
-    yAxisSvg!.width.baseVal.value + dotsSvg!.width.baseVal.value + 20;
-  const svgHeight =
-    dotsSvg!.height.baseVal.value + xAxisSvg!.height.baseVal.value;
+    heatmapWidth +
+    Y_AXIS_CHART_WIDTH_PX +
+    CONTENT_WRAPPER_LEFT_RIGHT_PADDING_PX * 2;
+
   const finalSvg = document.createElementNS(NAME_SPACE_URI, "svg");
+  applyAttributes(finalSvg, {
```

Comment:
using `applyAttributes` from svgUtils instead of using `setAttribute` multiple times